source/fuzz/transformation_inline_function.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2020 André Perez Maselco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_inline_function.h"
+
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationInlineFunction::TransformationInlineFunction(
+    const spvtools::fuzz::protobufs::TransformationInlineFunction& message)
+    : message_(message) {}
+
+TransformationInlineFunction::TransformationInlineFunction(
+    const std::map<uint32_t, uint32_t>& result_id_map,
+    uint32_t function_call_id) {
+  *message_.mutable_result_id_map() =
+      google::protobuf::Map<google::protobuf::uint32, google::protobuf::uint32>(
+          result_id_map.begin(), result_id_map.end());
+  message_.set_function_call_id(function_call_id);
+}
+
+bool TransformationInlineFunction::IsApplicable(
+    opt::IRContext* ir_context, const TransformationContext& /*unused*/) const {
```

Comment:
Some other things to be checked:
- The OpFunctionCall instruction should be the penultimate instruction in its block
- The block terminator should be OpBranch

This is important because:

(a) it means that the block is not a loop header or selection header, which avoids various validity problems
(b) it means that we can handle OpReturn / OpReturnValue instructions by turning them into OpBranch instructions to the successor of the original block

Reply:
Done.